### PR TITLE
Fix ubsan/tsan on Mac

### LINKED ISSUE #1913: tsan throws warnings on Mac
Probably false positives because of an uninstrumented libc++, but we should check that.

----
I also stumbled across the following [TSan flags](https://github.com/google/sanitizers/wiki/ThreadSanitizerFlags): 
* `force_seq_cst_atomics` (runtime)
* `-mllvm -tsan-instrument-atomics` (compiler)

Useful to us?

_Originally posted by @julianmenzler in https://github.com/hyrise/hyrise/pull/1909_

### PR DESCRIPTION
**Pull request by mrks**
fixes #1913.

### CODE REVIEW

File: resources/.ubsan-blacklist.txt [hunk 1]
```diff
@@ -6,4 +6,7 @@
 [undefined]
 # The hash map seems to reinterpret memory addresses in some semi-legal (but working) way. Example:
 # ska::bytell_hash_map<int, int> h; h.emplace(11, 11); std::cout << h[11] << std::endl;
-src:*bytell_hash_map.hpp
\ No newline at end of file
+src:*bytell_hash_map.hpp
+
+# A false positive in LZ4
```
Comment (Bensk1) [inline resources/.ubsan-blacklist.txt:11]: Are we sure that this is a false positive or could this also be an error in LZ4?

File: src/lib/CMakeLists.txt [hunk 2]
```diff
@@ -703,9 +703,15 @@ set_source_files_properties(
...
-# Add our jemalloc build as an interface library so that the build process of hyrise does not have to wait for it
-# -rdynamic tells the linker to export the library's symbols so that plugins can use them.
-target_link_libraries(hyrise_impl PUBLIC ${LIBRARIES} INTERFACE custom_jemalloc -rdynamic)
+target_link_libraries(hyrise_impl PUBLIC ${LIBRARIES} -rdynamic)
+
+if(NOT "${ENABLE_THREAD_SANITIZATION}")
+    # Add our jemalloc build as an interface library so that the build process of hyrise does not have to wait for it
+    # -rdynamic tells the linker to export the library's symbols so that plugins can use them. Do not use it for
+    # thread sanitization as tsan on Mac works better with its own allocator.
+    target_link_libraries(hyrise_impl INTERFACE custom_jemalloc -rdynamic)
```
[the remark this answers is not in the document]
Comment (mrks) [reply src/lib/CMakeLists.txt:712]: Should not be possible (unless jemalloc is broken). If anything, I'd expect this to be safer, as no external allocator gets in the way of ubsan's / tsan's instrumentation.

File: resources/.ubsan-blacklist.txt (hunk 1 above)
Comment (mrks) [reply resources/.ubsan-blacklist.txt]: Might also be an actual issue in LZ4, but nothing that we can fix. Added a link to https://github.com/lz4/lz4/issues/847

Comment (mrks) [reply resources/.ubsan-blacklist.txt]: Having seen a minimal example outside of LZ4, this is simply a new check that the LZ4 cannot work with. Quite sure that this is not a bug.

**Pull request merged**